pkg/controllers/updaterun/stop_test.go

```diff
@@ -0,0 +1,511 @@
+				},
+			},
+			wantFinished:   false,
+			wantError:      false,
+			wantAbortError: false,
+			wantProgressCond: metav1.Condition{
+				Type:               string(placementv1beta1.StageUpdatingConditionProgressing),
+				Status:             metav1.ConditionFalse,
+				ObservedGeneration: 1,
+				Reason:             condition.StageUpdatingStoppedReason,
+			},
+		},
+		{
+			name: "multiple clusters with mixed states",
+			updateRun: &placementv1beta1.ClusterStagedUpdateRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-updaterun",
+					Generation: 1,
+				},
+				Status: placementv1beta1.UpdateRunStatus{
+					DeletionStageStatus: &placementv1beta1.StageUpdatingStatus{
+						StageName: "deletion",
+						Clusters: []placementv1beta1.ClusterUpdatingStatus{
+							{
+								ClusterName: "cluster-1",
+								Conditions: []metav1.Condition{
+									{
+										Type:               string(placementv1beta1.ClusterUpdatingConditionStarted),
+										Status:             metav1.ConditionTrue,
+										ObservedGeneration: 1,
+										LastTransitionTime: now,
+										Reason:             condition.ClusterUpdatingStartedReason,
+									},
+									{
+										Type:               string(placementv1beta1.ClusterUpdatingConditionSucceeded),
+										Status:             metav1.ConditionTrue,
+										ObservedGeneration: 1,
+										LastTransitionTime: now,
+										Reason:             condition.ClusterUpdatingSucceededReason,
+									},
+								},
+							},
+							{
+								ClusterName: "cluster-2",
+								Conditions: []metav1.Condition{
+									{
+										Type:               string(placementv1beta1.ClusterUpdatingConditionStarted),
+										Status:             metav1.ConditionTrue,
+										ObservedGeneration: 1,
+										LastTransitionTime: now,
+										Reason:             condition.ClusterUpdatingStartedReason,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			toBeDeletedBindings: []placementv1beta1.BindingObj{
+				&placementv1beta1.ClusterResourceBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						DeletionTimestamp: &deletionTime,
+					},
+					Spec: placementv1beta1.ResourceBindingSpec{
+						TargetCluster: "cluster-2",
+					},
+				},
+			},
+			wantFinished: false,
+			wantError:    false,
+			wantProgressCond: metav1.Condition{
+				Type:               string(placementv1beta1.StageUpdatingConditionProgressing),
+				Status:             metav1.ConditionUnknown,
+				ObservedGeneration: 1,
+				Reason:             condition.StageUpdatingStoppingReason,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &Reconciler{}
+
+			gotFinished, gotErr := r.stopDeleteStage(tt.updateRun, tt.toBeDeletedBindings)
+
+			// Check finished result.
+			if gotFinished != tt.wantFinished {
+				t.Errorf("stopDeleteStage() finished = %v, want %v", gotFinished, tt.wantFinished)
+			}
+
+			// Check error expectations.
+			if tt.wantError {
```

Comment:
Lets check the error similar to TestStopUpdatingStage, i.e. check the error msg instead of checking to see if error is errStagedUpdatedAborted, if we construct the wantErr correctly we don't need to use errors.Is ?